compiler/rustc_borrowck/src/diagnostics/move_errors.rs

```diff
@@ -635,19 +629,13 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
                     self.suggest_cloning(err, place_ty, expr, Some(use_spans));
                 }
 
-                let mut path = None;
-                let ty = self.infcx.tcx.short_ty_string(place_ty, &mut path);
+                let ty = self.infcx.tcx.short_string(place_ty, err.long_ty_path());
```

Comment:
This is an example of the intended way of using this. This is the safest way of creating a short string because there's no way of forgetting to print the "full name written to `path`" message.